elastic/kibana
# [Actionable Observability] - Add alert details page feature flag by App

## Summary

It fixes #142718 by providing a feature flag-per-App (APM, Logs, Metrics, and Uptime). And it removes the previous global feature flag for the Alert details page `xpack.observability.unsafe.alertDetails.enabled`

## How to review
Besides the static code review, a manual test is needed, and here is how:
1. Create at least one rule from each App (APM, Logs, Metrics, and Uptime) that fires at least an alert
2. Add the following keys to your `kibana.yml` config file with `true` value.

   -  `xpack.observability.unsafe.alertDetails.apm.enabled: ture`
   -  `xpack.observability.unsafe.alertDetails.metrics.enabled: true`
   -  `xpack.observability.unsafe.alertDetails.logs.enabled: true`
   -  `xpack.observability.unsafe.alertDetails.uptime.enabled: true`

3. From the alert table click on View alert details
      <img width="224" alt="Screenshot 2022-10-11 at 13 37 39" src="https://user-images.githubusercontent.com/6838659/195079934-1d04f509-c4ac-41e1-81ba-415bd0cff18b.png">
4.  You should see the Alert details page 
   - Repeat it on different rule types alert 
5. Click on the alert reason message, which will open the alert flyout out
   <img width="842" alt="Screenshot 2022-10-11 at 13 40 33" src="https://user-images.githubusercontent.com/6838659/195080426-a94f4320-e8f1-4b04-9d02-181b40839a98.png">
6. You should see the button View alert details
   <img width="294" alt="Screenshot 2022-10-11 at 13 42 49" src="https://user-images.githubusercontent.com/6838659/195080997-6b6f182a-a0e1-4686-bd45-4649ec30ad1e.png">
7. Change the value of the feature flags to `false` (step 2) and retry the steps again.
   - When the feature flag is `false` for a given App you should see the alert flyout in step 3, instead of the Alert details page for that specific App only.
   - When the feature flag is `false` for a given App you should NOT see the Alert details button in step 6  for that specific App only.
### Checklist

Delete any items that are not applicable to this PR.
